git-sidecar/rootfs/clone.sh

```diff
@@ -5,10 +5,27 @@ set -x
 : "${VCS_LOCAL_PATH:=/src}"
 : "${VCS_REPO:?}"
 : "${VCS_REVISION:?}"
+: "${GITHUB_PULL_REQUEST:=}"
+: "${GITHUB_REF:=}"
+: "${GITHUB_REF_TYPE:=}"
 
-git clone --depth=50 "${VCS_REPO}" "${VCS_LOCAL_PATH}"
-cd "${VCS_LOCAL_PATH}"
-
-git fetch origin "${VCS_REVISION}"
-git checkout -qf FETCH_HEAD
-git reset --hard -q "${VCS_REVISION}"
+if [ "$GITHUB_PULL_REQUEST" ]; then
+    git clone --depth=50 "${VCS_REPO}" "${VCS_LOCAL_PATH}"
+    cd "${VCS_LOCAL_PATH}"
+    git fetch origin "+refs/pull/${GITHUB_PULL_REQUEST}/merge:"
```

Comment:
Hi! Thanks for the enhancement!

Would this change still allows us to build not only "the commit w/ the pull-request merged into the base branch" but also "the commit which is HEAD of the pull-request"?

I'm wondering if this breaks the latter - which I believe is the existing behavior.